cmd/common/registry.go

```diff
@@ -0,0 +1,342 @@
+				formatter.SetShowColumns(showColumns)
+				runner.Columns().SetErrorCallback(fe.Error)
+
+				// Wire up callbacks before handing over to runtime depending on the output mode
+				switch outputMode {
+				default:
+					formatter.SetEventCallback(fe.Output)
+
+					// Enable additional output, if the gadget supports it (e.g. profile/cpu)
+					//  TODO: This can be optimized later on
+					formatter.SetEnableExtraLines(true)
+
+					runner.Columns().SetEventCallback(formatter.EventHandlerFunc())
+					if gadget.Type().IsPeriodic() {
+						// In case of periodic outputting gadgets, this is done as full table output, and we need to
+						// clear the screen for every interval, that's why we add fe.Clear here
+						runner.Columns().SetEventCallbackArray(formatter.EventHandlerFuncArray(
+							fe.Clear,
+							func() {
+								fe.Output(formatter.FormatHeader())
+							},
+						))
+						break
+					}
+					fe.Output(formatter.FormatHeader())
+					runner.Columns().SetEventCallbackArray(formatter.EventHandlerFuncArray())
+				case utils.OutputModeJSON:
+					runner.Columns().SetEventCallback(printEventAsJSON)
+					runner.Columns().SetEventCallbackArray(printEventAsJSON)
+				}
+			} else {
+				defer func() {
+					res, _ := runner.GetResult()
+					switch outputMode {
+					default:
+						transformer := gadget.(gadgets.GadgetOutputFormats)
+						formats, defaultFormat := transformer.OutputFormats()
+						transformed, _ := formats[defaultFormat].Transform(res)
+						fmt.Fprint(os.Stdout, string(transformed))
+					case utils.OutputModeJSON:
+						fmt.Fprint(os.Stdout, string(res))
+					}
+				}()
+			}
+
+			// Finally, hand over to runtime
+			return runner.RunGadget(runtimeParams, operatorsParamCollection, operatorsPerGadgetParamCollection, params)
+		},
+	}
+
+	if gadget.Type() != gadgets.TypeOneShot {
+		// Add timeout
+		cmd.PersistentFlags().IntVarP(&timeout, "timeout", "t", 0, "Number of seconds that the gadget will run for, 0 to disable")
+	}
+
+	cmd.PersistentFlags().BoolVarP(
+		&verbose,
+		"verbose", "v",
+		false,
+		"Print debug information",
+	)
+
+	outputFormats.Append(gadgets.OutputFormats{
+		"json": {
+			Name:        "JSON",
+			Description: "The output of the gadget is returned as raw JSON",
+			Transform:   nil,
+		},
+	})
+	defaultOutputFormat = "json"
+
+	// Add output flags
+	if columns != nil {
+		outputFormats.Append(gadgets.OutputFormats{
+			"columns": {
+				Name:        "Columns",
+				Description: "The output of the gadget is formatted in human readable columns",
+				Transform:   nil,
+			},
+		})
+		defaultOutputFormat = "columns"
+
+		cmd.PersistentFlags().StringSliceVarP(
+			&showColumns,
+			"columns", "C",
+			columns.GetDefaultColumns(),
+			"Columns to output",
+		)
+		cmd.PersistentFlags().StringSliceVarP(
+			&filters,
+			"filter", "F",
+			[]string{},
+			"Filter rules",
+		)
+
+		// Add params matching the gadget type
+		params.AddParams(gadgets.GadgetParams(gadget, columns))
```

Comment:
The `AddParam()` is a helper function that just adds a key/value pair - without all the fancy schmancy of the Params package. But now that I think about it, maybe that (`AddParam()`) shouldn't be necessary in the first place (more on that below). And then it could just be named `Add(params ...Param)`.

Reply:
Naming should be better now, wdyt?

Params now has Add(...) and AddKeyValuePair(key, value).